twilio/http/http_client.py

```diff
@@ -65,7 +66,7 @@ def request(self, method, url, params=None, data=None, headers=None, auth=None,
         response = session.send(
             prepped_request,
             allow_redirects=allow_redirects,
-            timeout=timeout,
+            timeout=timeout if timeout is not None else self.timeout,
```

Comment:
nit: simplify to `timeout=timeout if timeout else self.timeout`, since `if timeout` evaluates to `True` when it is not `None`.

Reply:
Not sure what passing a timeout of `0` would do, but that would result in different behavior in the suggestion.

Reply:
Just tested this, `timeout = 0` evaluates to `False`, then `timeout` would take on the value of `self.timeout`. This seems like desired behavior?

Reply:
Not sure what's correct or desired. Should setting the timeout to 0 result in not using `self.timeout` and essentially disabling the timeout altogether? Mainly just noting that `if timeout is not None` is not the same as `if timeout` (things like this have bitten me in the past).

Reply:
Would it be worth doing something like `timeout=0` by default rather than `None` then, and checking for the 0 - if it's 0, use the `self.timeout` value to avoid the `ValueError` exception in urllib3. Otherwise, use the `timeout` value?

`def request(self, method, url, params=None, data=None, headers=None, auth=None, timeout=0, allow_redirects=False):
    if timeout == 0:
        timeout = self.timeout
    # Some code
    response = session.send(
            prepped_request,
            allow_redirects=allow_redirects,
            timeout=timeout
        )`

Would also require a similar check on the value of self.timeout, potentially in the __init__, to make sure the user isn't setting self.timeout to be 0

Reply:
I think it's fine as-is. Don't need to over-engineer for the case that the client has a default timeout and a request wants to _remove_ the timeout. There should really always be _some_ timeout.